Dangerfile

```diff
@@ -22,13 +22,25 @@ changed_files = git.diff.stats[:files]
 excluded_changed_files = changed_files.select { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
 filtered_changed_files = changed_files.reject { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
 
+# fetch master for diff comparison
+`git fetch origin master`
+
+# get branch name
+branches = `git branch`
+current_branch = branches.match(/\*\s+([^\s]+)/m)[1]
+
 # ignores whitespace for the purpose of determining lines of code changed
-changes = `git diff master... -w --stat`.split("\n")
-lines_of_code = changes.sum(0) do |change|
-  if change == changes.last || EXCLUSIONS.any? { |exclusion| change.match(exclusion) }
-    0
-  else
-    change.match(/\|\s+(\d+)/)[1].to_i
+diff_changes = `git diff master..#{current_branch} -w --stat`.split("\n")
```

Comment:
```suggestion
diff_changes = `git diff master...#{current_branch} -w --stat`.split("\n")
```

This should be 3 dots here.

Reply:
I'm kind of fuzzy on this. I had initially had it as 3 dots but changed it back to two after reading some explanations about the two. My understanding from what I read was that using 3 dots does a diff comparison between your branch and where it branched off from master as opposed to where master currently is and that 2 dots does the diff comparison between your current branch and where master is currently.

Reply:
Three dot diff is what GitHub shows in the "Files Changed" tab.

I think we should be counting what's visible in that tab instead of something hidden.

Also, I believe if master changed significantly it's going to be counting the diffs from master. Say master deleted 1000 lines of code. Well now the current branch has a 1000 line lead in the line count race.

Take this branch for example, run this command:

```
❯ git diff origin/master..tjh-dangerfile --stat
...
128 files changed, 1208 insertions(+), 215727 deletions(-)
```
Now run it with three dots:
```
❯ git diff origin/master...tjh-dangerfile --stat
Dangerfile | 31 +++++++++++++++++++++----------
1 file changed, 21 insertions(+), 10 deletions(-)
```
That's the same as what GitHub shows.